File: python3/twoSum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        nums_len = len(nums)
        d = dict()
        for i in range(nums_len):
            d[nums[i]] = i
        for i in range(nums_len):
            if d.get(target - nums[i]):
                return i, d[target - nums[i]]
        
s = Solution()
assert(s.twoSum([2, 7, 11, 5], 9) == (0, 1))
assert(s.twoSum([2, 5, 5, 11], 10) == (1, 2))

File: javascript/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  const numMap = {}
  const len = nums.length

  for (let i=0; i<len; i++) {
    numMap[nums[i]] = i
  }
  for (let i=0; i<len; i++) {
    if (numMap[target-nums[i]]) {
      return [i, numMap[target-nums[i]]]
    }
  }
};

const { assertArrayEqual } = require('./assert')
assertArrayEqual(twoSum([1, 2, 3, 4], 6), [1, 3])
